compiler/test/Transforms/shapeReification.mlir

```diff
@@ -62,6 +62,24 @@ func.func @gelu(%arg0: tensor<?x3072xf32>) -> tensor<?x3072xf32> {
   return %0 : tensor<?x3072xf32>
 }
 
+func.func @tf_StridedSlice(%arg0: tensor<?x5x64xf16>) -> tensor<?x5x1x64xf16> {
+  %0 = mhlo.constant dense<1> : tensor<4xi32>
+  %1 = mhlo.constant dense<0> : tensor<4xi32>
+  %2 = mhlo.custom_call @tf.StridedSlice(%arg0, %1, %1, %0) {backend_config = "", byteir_attrs = {begin_mask = 11 : i64, ellipsis_mask = 0 : i64, end_mask = 11 : i64, new_axis_mask = 4 : i64, shrink_axis_mask = 0 : i64}} : (tensor<?x5x64xf16>, tensor<4xi32>, tensor<4xi32>, tensor<4xi32>) -> tensor<?x5x1x64xf16>
```

Comment:
Is there any opportunity to lower `tf.StridedSlice` to `mhlo.real_dynamic_slice`?

Reply:
We will replace tf.StridedSlice with milo.real_dynamic_slice in tf frontend, the reason why we can't replace all of them is that tf.StridedSlice is more complex than mhlo.real_dynamic_slice, including the following points:
1. begin, end, stride of tf.StridedSlice can be negative, and the length does not have to be equal to the rank of input.
2. in addition to begin, end, stride, output size is also affected by begin_mask, end_mask
3. ellipsis_mask, new_axis_mask, shrink_axis_mask attributes are very flexible and will affect the size and rank of output.